packages/expo-updates/android/src/main/java/expo/modules/updates/launcher/DatabaseLauncher.kt

```diff
@@ -97,25 +100,39 @@ class DatabaseLauncher(
     val embeddedUpdate = EmbeddedManifestUtils.getEmbeddedUpdate(context, configuration)
     val extraHeaders = FileDownloader.getExtraHeadersForRemoteAssetRequest(launchedUpdate, embeddedUpdate?.updateEntity, launchedUpdate)
 
-    val launchAssetFile = ensureAssetExists(launchAsset, database, embeddedUpdate, extraHeaders)
+    val embeddedLaunchAsset = if (!shouldCopyEmbeddedAssets) {
+      embeddedUpdate?.assetEntityList
+        ?.find { it.key == launchAsset.key }
+        ?.embeddedAssetFilename
+        ?.let {
+          // react-native uses `assets://` to indicate loading a bundle from assets
+          "assets://$it"
+        }
+    } else {
+      null
+    }
+    val launchAssetFile = embeddedLaunchAsset ?: ensureAssetExists(launchAsset, database, embeddedUpdate, extraHeaders)
     if (launchAssetFile != null) {
       this.launchAssetFile = launchAssetFile.toString()
     }
 
     val assetEntities = database.assetDao().loadAssetsForUpdate(launchedUpdate!!.id)
+    val isEmbeddedLaunch = embeddedUpdate != null && launchedUpdate?.id?.equals(embeddedUpdate.updateEntity.id) ?: false
 
     localAssetFiles = embeddedAssetFileMap().apply {
       for (asset in assetEntities) {
         if (asset.id == launchAsset.id) {
           // we took care of this one above
           continue
         }
-        val filename = asset.relativePath
-        if (filename != null) {
+        val filename = asset.relativePath ?: continue
+        if (!isEmbeddedLaunch || shouldCopyEmbeddedAssets) {
```

Comment:
i serve `file:///android_res/` only when `isEmbeddedLaunch=false`. that is based on the assumption that eas updates will upload all assets. we don't have asset exclusion that reuse embedded assets. i hope the assumption is correct

Reply:
removed this incorrect assumption. now we also support asset selection